apisix/plugins/openfunction.lua

```diff
@@ -0,0 +1,36 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local ngx_encode_base64 = ngx.encode_base64
+local plugin_name, plugin_version, priority = "openfunction", 0.1, -1902
+
+local openfunction_authz_schema = {
+    service_token = {type = "string"}
+}
+
+local function request_processor(conf, ctx, params)
+    local headers = params.headers or {}
+    -- setting authorization headers if not already set
+    if not headers["authorization"] and conf.authorization
```

Comment:
What if users configure an auth plugin with the OpenFunction plugin, and the authorization header from client is not designed for the FaaS. In such a case, the authentication will fail.

Reply:
Yes,it is a problem.
I think it can be avoided  by overwriting authorization headers only when 'conf.authorization.service_Token' is given by users ,since auth plugin runs before OpenFcuntion plugin. 
However,it is still hard for user to apply user-specific authorization header when auth plugin runs.Will it be a good way to put user-specific token in an other header(e.g. 'o-function-token') and convert it to authorization header in access step?

Reply:
@jackkkkklee Just set the authorization header once users specify it in plugin config. `headers` variable is used only by this plugin and if you change it, the original request headers won't be changed, so there is no side-effect.